README.md

```diff
@@ -344,3 +344,127 @@ Peeps.new(id: result[0]['id'], message: result[0]['message'],time: result[0]['ti
  end
 6) Editing index.erb to include Time
 <li><%= peep.time %></li>
+
+# STEP 5: USER STORY 4
+As a Maker
+So that I can post messages on Chitter as me
+I want to sign up for Chitter
+1) PSQL & db/migrations
+CREATE TABLE users (id SERIAL PRIMARY KEY, email VARCHAR(100) UNIQUE, password VARCHAR(100),name VARCHAR(100), username VARCHAR(50) UNIQUE);
+2) Update Setup_test_Database to truncate users table as well
+connection.exec("TRUNCATE peep, users;")
+3) Feature test
+feature 'Sign up' do
+  scenario 'A user can sign up to Chitter' do
+    visit '/users/new'
+    fill_in('email',with: 'example@example.com')
+    fill_in('password',with: 'example123')
+    fill_in('name',with: 'Example Surname')
+    fill_in('username',with: 'exampleusername')
+    click_button('Submit')
+
+    expect(page).to have_content "Welcome, Example Surname"
+  end
+end
+4) app.rb to implement get route
+get '/users/new' do
+  erb: 'users/new'
+end
+5) erb: new
+<form action="/users" method="post">
+  <input type="email" name="email" />
+  <input type="password" name="password" />
+  <input type="text" name="name" />
+  <input type="text" name="username" />
+  <input type="submit" value="Submit" />
+</form>
+6) app.rb to implement post route
+post '/users' do
+  User.create(email: params[:email],password: params[:password], name: params[:name], username: params[:username])
+  redirect '/peeps'
+end
+7)user_spec test
+require 'user'
+require 'database_helpers'
+
+describe User do
+  describe '.create' do
+    it 'creates a new user' do
+      user = User.create(email: 'example@example.com',password: 'example123',name: 'Example Surname', username: 'exampleusername')
+
+      expect(peep.email).to eq "example@example.com"
+      expect(peep.password).to eq "example123"
+      expect(peep.name).to eq "Example Surname"
+      expect(peep.username).to eq "Example Username"
+    end
+  end
+8) implementation of User class
+class User
+  def self.create(email:,password:,name:,username:)
+    result = DatabaseConnection.query("INSERT into users (email,password,name,username)
+    VALUES ('#{email}', '#{password}','#{name}','#{username}') RETURNING email, password, name, username;")
+    User.new(email: result[0]['email'],password: result[0]['password'],name: result[0]['name'],username: result[0]['username'])
```

Comment:
Just wondering whether we need to have user password accessible via attribute reader (and, for that matter, require it to initialise it to create a User class instance) -- perhaps, just keeping it in the database in encrypted form is sufficient for subsequent authentication?